datasetinsights/io/kfp_output.py

```diff
@@ -68,3 +76,15 @@ def write_metric(self):
         logger.debug(
             f"Metrics file {self.filename} saved at path:" f" {self.filepath}"
         )
+
+    def create_tb_visualization_json(self):
+        try:
+            metadata = {
+                "outputs": [{"type": "tensorboard", "source": self.tb_log_dir}]
+            }
+            with open("/mlpipeline-ui-metadata.json", "w") as f:
+                json.dump(metadata, f)
+
+        # when we don't have write permission
+        except IOError:
```

Comment:
If the output json path is not hard coded, this check can be removed. It's not very good practice if this failed silently.

Reply:
The output path has to be hardcoded.

Reply:
Also, the reason for this was when we run this locally and we don't have write permission on `/` which will cause it to fail which we ideally don't want as writing this file is not necessary when running locally instead of kubeflow.